query/src/main/scala/filodb/query/exec/StitchRvsExec.scala

```diff
@@ -12,41 +12,55 @@ import filodb.query.Query.qLogger
 
 object StitchRvsExec {
 
-  def stitch(v1: RangeVector, v2: RangeVector): RangeVector = {
-    val rows = StitchRvsExec.merge(Seq(v1.rows(), v2.rows()))
-    IteratorBackedRangeVector(v1.key, rows, RvRange.union(v1.outputRange, v2.outputRange))
-  }
+  private class RVCursorImpl(vectors: Iterable[RangeVectorCursor], outputRange: Option[RvRange])
+    extends RangeVectorCursor {
+    private val bVectors = vectors.map(_.buffered)
+    val mins = new mutable.ArrayBuffer[BufferedIterator[RowReader]](2)
+    val nanResult = new TransientRow(0, Double.NaN)
+    val tsIter: BufferedIterator[Long] = outputRange match {
+      case Some(RvRange(startMs, stepMs, endMs)) =>
+        Iterator.iterate(startMs){_ + stepMs}.takeWhile( _ <= endMs).buffered
+      case None                                  => Iterator.iterate(Long.MaxValue)(_ =>Long.MaxValue).buffered
```

Comment:
Is there a reason not to error out if outputRange is None ?
I think we should, or else, bugs may silently creep in.